www/include/common/javascript/datepicker/localizedDatepicker.js

```diff
@@ -132,58 +130,100 @@ function setUserFormat() {
 
+/**
+ * Turn the event Off to avoid infinite loop.
+ */
 function turnOffEvents() {
     $(".datepicker").off('change');
     $(".timepicker").off('change');
 }
 
-function updateEndTime() {
+/**
+ * Update the end datepicker and timepicker according to the start values.
+ */
+function updateDateAndTime() {
     let start = moment($('[name="alternativeDateStart"]').val()
-        + ' ' +  $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
     let end = moment($('[name="alternativeDateEnd"]').val()
-        + ' ' +  $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
 
     if (start.isAfter(end) || start.isSame(end)) {
         turnOffEvents();
         start.add($('#duration').val(), $('#duration_scale').val());
         $(".datepicker").last().datepicker("setDate", start.format($(".datepicker").last().datepicker(
             "option",
             "dateFormat"
-        ).toUpperCase().replace(/Y/g,'YY')));
+        ).toUpperCase().replace(/Y/g, 'YY')));
         $(".timepicker").last().timepicker("setTime", start.format("HH:mm"));
         turnOnEvents();
     }
 }
 
-function updateStartTime() {
+/**
+ * Used for the end DATEPICKER date.
+ * Update the end timepicker according to the start values.
+ */
+function updateEndTime() {
     let start = moment($('[name="alternativeDateStart"]').val()
-        + ' ' +  $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
     let end = moment($('[name="alternativeDateEnd"]').val()
-        + ' ' +  $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
 
     if (start.isAfter(end) || start.isSame(end)) {
         turnOffEvents();
-        end.subtract($('#duration').val(), $('#duration_scale').val());
-        $(".datepicker").first().datepicker("setDate", end.format($(".datepicker").first().datepicker(
-            "option",
-            "dateFormat"
-        ).toUpperCase().replace(/Y/g,'YY')));
-        $(".timepicker").first().timepicker("setTime", end.format("HH:mm"));
+        start.add($('#duration').val(), $('#duration_scale').val());
+        $(".timepicker").last().timepicker("setTime", start.format("HH:mm"));
         turnOnEvents();
     }
 }
+
+/**
+ * Used for the end TIMEPICKER time.
+ * Update the end timepicker according to the start values and the chosen duration,
+ * and display a warning to the user.
+ */
+function checkEndTime() {
+    let startTime = $(".timepicker").first().val()
+    let start = moment($('[name="alternativeDateStart"]').val()
+        + ' ' + startTime, "MM/DD/YYYY HH:mm");
+
+    let endTime = $(".timepicker").last().val();
+    let end = moment($('[name="alternativeDateEnd"]').val()
+        + ' ' + endTime, "MM/DD/YYYY HH:mm");
+
+    if (start.isAfter(end) || start.isSame(end)) {
```

Comment:
you could use `isSameOrAfter` : https://momentjscom.readthedocs.io/en/latest/moment/05-query/05-is-same-or-after/

Reply:
done